dnf.spec

```diff
@@ -85,6 +85,8 @@ Requires:       python3-%{name} = %{version}-%{release}
 %if 0%{?rhel} && 0%{?rhel} <= 7
 Requires:       python-dbus
 Requires:       %{_bindir}/sqlite3
+%elif 0%{?fedora}
+Recommends:     (%{_bindir}/sqlite3 if bash-completion)
```

Comment:
I am proposing to add additional condition `(%{_bindir}/sqlite3 if (bash-completion and python3-dnf-plugins-core))`.
If `python3-dnf-plugins-core` is not installed then the database cannot be updated and bash completion provides outdated results.

Be honest this approach is not perfect, because it uses resources during RPM transaction. DNF5 uses an alternative approach without external database.